src/model/codec-info.ts

```diff
@@ -84,11 +106,20 @@ export class CodecInfo implements SdpBlock {
     this.feedback.forEach((fb) => {
       lines.push(new RtcpFbLine(this.pt, fb));
     });
-    // Now all Fmtp
-    this.fmtParams.forEach((fmt) => {
-      lines.push(new FmtpLine(this.pt, fmt));
-    });
-
+    if (this.fmtParams.length > 0) {
```

Comment:
I think we should do this at the `FmtpLine` level, not the `CodecInfo` level: it should still be possible to serialize multiple fmtp lines.

What we want is for the caller (wcme) to be able to grab the existing fmtp lines in the `CodecInfo`, and 'expand' one of them by adding extra params.

Reply:
> I think we should do this at the FmtpLine level, not the CodecInfo level: it should still be possible to serialize multiple fmtp lines.
Since the `fmtParams` in `CodecInfo` currently is defined as `Array` type, so we can not prevent upper layer adding new fmtp params through the `push` method, and that actually the way we used to add new fmtp params in WCME as this [code](https://sqbu-github.cisco.com/CPaaS/web-client-media-engine/blob/b1e8363cd79b1bbafba6176531e3fb727cf5e187/src/multistream-connection.ts#L843) shows. 
So what's this PR want to solve is:  when there are multiple fmtp lines with the same payload type in `fmtParams`, just merge it as one line. 

> What we want is for the caller (wcme) to be able to grab the existing fmtp lines in the CodecInfo, and 'expand' one of them by adding extra params.

Yes, of course we can do this, but if we keep the `fmtParams` still as `Array` type,  we still need to consider the problem that the `fmtParams` may be contains multiple values(lines) with same payload type, which would result in Firefox failed.

Reply:
Before the `push` method being called, there may be only 1 item in `fmtParams` , which could be:

> ['fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f']

After the `push` method being called like this:

> ci.fmtParams.push(`max-mbps=${defaultMaxVideoEncodeMbps}`);
> ci.fmtParams.push(`max-fs=${defaultMaxVideoEncodeFrameSize}`);

Then the value of `fmtParams` would contains 3 items, which could be: 

> ['fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f', max-mbps=244800, max-fs=8160]

What this PR want to solve is, merge all items in the `fmtParams` , and make it  as one line, which final be:

> ['fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f';max-mbps=244800;max-fs=8160]

Reply:
Ok, I re-read the spec and [it](https://datatracker.ietf.org/doc/html/rfc4566#section-6) _does_ say only one `fmtp` line is allowed per payload type, so since `CodecInfo` is on the scope of a payload type, it does make sense to make sure we always serialize to a single fmtp line.  But this just means that, when serializing `CodecInfo`, instead of writing a new fmtp line for each param, we just need to use a single line for all the params?

This doesn't protect against multiple, duplicate values, though.  But if we want to protect against that, `FmtpLine` should parse the params into an object/map, instead of a single string.  `CodecInfo` should then hold a map of these params, instead of a list.  So then the codec can add a new param just as before, but can avoid duplicate param values.

Reply:
Code update.